pkg/services/login/loginservice/loginservice.go

```diff
@@ -16,9 +16,25 @@ var (
 	logger = log.New("login.ext_user")
 )
 
+type Service interface {
+	UpsertUser(ctx context.Context, cmd *models.UpsertUserCommand) error
+}
+
+type Store interface {
```

Comment:
We might have a tad double work, I also did a part here https://github.com/grafana/grafana/pull/44907

Reply:
Happy to wait until your PR is merged, this one has failing tests anyway, then I'll rebase